api/src/test/java/org/openmrs/util/databasechange/Database2_2_0UpgradeIT.java

```diff
@@ -0,0 +1,115 @@
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.openmrs.util.DatabaseUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.openmrs.util.OpenmrsUtil;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests database upgrade from OpenMRS 2.2.0.
+ */
+public class Database2_2_0UpgradeIT extends BaseContextSensitiveTest {
+	
+	public static final String TEST_DATA_DIR = "/org/openmrs/util/databasechange/";
+	
+	public static final String UPGRADE_TEST_2_1_0_TO_2_20_DATASET = TEST_DATA_DIR
+		+ "database2_1To2_20UpgradeTest-dataSet.xml";
+	
+	public static final String STANDARD_TEST_2_2_0_DATASET = TEST_DATA_DIR + "standardTest-2.2.0-dataSet.xml";
+	
+	public final static String DATABASE_PATH = TEST_DATA_DIR + "openmrs-2.2.0.h2.db";
+	
+	private DatabaseUpgradeTestUtil upgradeTestUtil;
+	
+	private static File testAppDataDir;
+	
+	@BeforeClass
+	public static void beforeClass() throws IOException {
+		testAppDataDir = File.createTempFile("appdir-for-unit-tests", "");
+		testAppDataDir.delete();// so we can make turn it into a directory
+		testAppDataDir.mkdir();
+		
+		System.setProperty(OpenmrsConstants.APPLICATION_DATA_DIRECTORY_RUNTIME_PROPERTY, testAppDataDir.getAbsolutePath());
+		OpenmrsUtil.setApplicationDataDirectory(testAppDataDir.getAbsolutePath());
+	}
+	
+	@AfterClass
+	public static void afterClass() throws Exception {
+		FileUtils.deleteDirectory(testAppDataDir);
+		//Just to be safe, not to affect other units in the test suite
+		System.clearProperty(OpenmrsConstants.APPLICATION_DATA_DIRECTORY_RUNTIME_PROPERTY);
+	}
+	
+	@Before
+	public void before() throws IOException, SQLException {
+		upgradeTestUtil = new DatabaseUpgradeTestUtil(DATABASE_PATH);
+	}
+	
+	@After
+	public void after() throws SQLException {
+		upgradeTestUtil.close();
+	}
+	
+	/**
+	 * This method creates mock order entry upgrade file
+	 *
+	 * @see org.openmrs.util.UpgradeUtil#getConceptIdForUnits(String)
+	 */
+	public static void createOrderEntryUpgradeFileWithTestData(String propString) throws IOException {
+		Properties props = new Properties();
+		props.load(new StringReader(propString));
+		String appDataDir = OpenmrsUtil.getApplicationDataDirectory();
+		File propFile = new File(appDataDir, DatabaseUtil.ORDER_ENTRY_UPGRADE_SETTINGS_FILENAME);
+		props.store(new FileWriter(propFile), null);
+		propFile.deleteOnExit();
+	}
+	
+	@Test
+	public void shouldHaveUpdatedOrderTableInDatabase() throws Exception {
+		String FULFILLER_STATUS = "RECEIVED";
+		String FULFILLER_COMMENT = "Comment";
+		
+		upgradeTestUtil.executeDataset(STANDARD_TEST_2_2_0_DATASET);
```

Comment:
What is the use of this? My guess is that the database is already at 2.1.x version and you should only be running the  upgrade script to get it to 2.2.0

Reply:
@wluyima  Are you refering to this line?:
`		upgradeTestUtil.executeDataset(UPGRADE_TEST_2_1_0_TO_2_20_DATASET);
`

I am not sure about this. I copied these lines from my previous created testcase in the 1.9 integration test.
If I do not use this line I am getting the following error. That's why I was keeping it inside.

`java.lang.IllegalArgumentException: Global property not found order.drugDosingUnitsConceptUuid:`


Any suggestions? Like I said earlier. The requirement of having such an upgrade test came up but I have not so much experience in this field